src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [, , , initialSeparator] = fromFormat;
+  const [, , , resultingSeparator] = toFormat;
+  const initialArray = fromFormat.slice(0, 3);
+  const resultingArray = toFormat.slice(0, 3);
+  const dateArray = date.split(initialSeparator);
+  const result = [];
+
+  for (const initialElement of initialArray) {
+    for (const resultingElement of resultingArray) {
+      const initialIndex = initialArray.indexOf(initialElement);
+      const resultingIndex = resultingArray.indexOf(resultingElement);
```

Comment:
Similar to the previous comment, calling `indexOf` inside a loop for `resultingArray` is unnecessary and inefficient. You should establish a direct mapping between the `initialArray` and `resultingArray` outside of the loop to avoid repeated searches.